src/netex_idf/stops.rs

```diff
@@ -137,49 +129,25 @@ fn stop_point_parent_id(
         .map(|stop_area| stop_area.id.clone()))
 }
 
-fn accessibility(quay: &Element) -> Option<StopAccessibility> {
-    fn availability(val: &str) -> Availability {
-        match val {
-            "true" => Availability::Available,
-            "false" => Availability::NotAvailable,
-            _ => Availability::InformationNotAvailable,
-        }
-    }
-
-    let accessibility_node = quay.only_child("AccessibilityAssessment")?;
-    let mobility_impaired_access = accessibility_node
-        .only_child("MobilityImpairedAccess")?
-        .text();
-    let limitation = accessibility_node
-        .only_child("limitations")?
-        .only_child("AccessibilityLimitation")?;
-    let visual_signs_available = limitation.only_child("VisualSignsAvailable")?.text();
-    let audio_signs_available = limitation.only_child("AudibleSignalsAvailable")?.text();
-
-    Some(StopAccessibility {
-        wheelchair_boarding: availability(&mobility_impaired_access),
-        visual_announcement: availability(&visual_signs_available),
-        audible_announcement: availability(&audio_signs_available),
-    })
-}
-
-fn get_or_create_equipment<'a>(
+pub fn get_or_create_equipment<'a>(
     quay: &Element,
-    equipments: &'a mut HashMap<StopAccessibility, Equipment>,
+    equipments: &'a mut HashMap<Accessibility, Equipment>,
     id_incr: &mut u8,
 ) -> Option<&'a mut Equipment> {
-    let accessibility = accessibility(quay)?;
+    let accessibility_node = quay.only_child("AccessibilityAssessment")?;
+    let accessibility = accessibility(accessibility_node)?;
 
-    let equipment = equipments.entry(accessibility).or_insert_with(|| {
-        let StopAccessibility {
-            wheelchair_boarding,
+    let equipment = equipments.entry(accessibility.clone()).or_insert_with(|| {
+        let Accessibility {
+            wheelchair,
```

Comment:
You can also destructure with a different name for the variable.
```rust
let Accessibility {
            wheelchair: wheelchair_boarding,
            visual_announcement,
            audible_announcement,
        } = accessibility;
```
This way, you do not need to do the swap of names below on line 150. I know it's highly subjective but I do prefer to name correctly the variable early.